File: src/test/java/br/ufsc/bridge/platform/validation/rules/CepTest.java
package br.ufsc.bridge.platform.validation.rules;

import org.junit.Assert;
import org.junit.Test;

import br.ufsc.bridge.platform.validation.ValidationTest;

public class CepTest extends ValidationTest {

	private static final CepRule RULE = Rules.cep;

	@Test
	public void testarRequired() {
		Assert.assertNull("Cep não deve ser requerido por padrão", this.validate(null, RULE));
	}

	@Test
	public void cepDeveConterExatamente8Digitos() {
		Assert.assertNull("CEP com 8 dígitos é válido", this.validate("88106580", RULE));

		Assert.assertNotNull("CEP deve conter exatamente 8 digítos", this.validate("881065805", RULE));

		Assert.assertNotNull("CEP deve conter exatamente 8 digítos", this.validate("881", RULE));

		Assert.assertNotNull("CEP deve conter exatamente 8 digítos", this.validate("88106ABC", RULE));
	}
}

File: src/test/kotlin/br/ufsc/bridge/platform/validation/rules/CepTest.kt
package br.ufsc.bridge.platform.validation.rules

import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Assertions.assertNull
import org.junit.jupiter.api.Test

class CepTest {

    @Test
    fun `testar required`() {
        assertNull(Rules.cep.validate(null), "Cep não deve ser requerido por padrão")
    }

    @Test
    fun `deve conter exatamente 8 digitos`() {
        assertNull(Rules.cep.validate("88106580"), "CEP com 8 dígitos é válido")

        assertNotNull(Rules.cep.validate("881065805"), "CEP deve conter exatamente 8 digítos")

        assertNotNull(Rules.cep.validate("881"), "CEP deve conter exatamente 8 digítos")

        assertNotNull(Rules.cep.validate("88106ABC"), "CEP deve conter exatamente 8 digítos")
    }

    companion object {
        private val RULE = Rules.cep
    }
}
